go-concourse/concourse/internal/connection_test.go

```diff
@@ -381,6 +381,31 @@ var _ = Describe("ATC Connection", func() {
 				})
 			})
 
+			Describe("403 response", func() {
```

Comment:
Not a big deal but this will require a little bit of care when merging in combination with: https://github.com/concourse/concourse/pull/9010

In other PR, I am removing the test just above that is very similar to the one I am adding here. The difference being: one uses `HTTPAgent` client and the other the `connection` client.